fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectColumnsRule.java

```diff
@@ -53,16 +50,14 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         }));
 
         if (newMap.isEmpty()) {
-            List<ColumnRefOperator> outputColumns =
-                    projectOperator.getOutputColumns(new ExpressionContext(input.inputAt(0))).getStream().
-                            mapToObj(context.getColumnRefFactory()::getColumnRef).collect(Collectors.toList());
-            if (CollectionUtils.isNotEmpty(outputColumns)) {
-                ColumnRefOperator smallestColumn = Utils.findSmallestColumnRef(outputColumns);
+            ColumnRefSet candidates = new ColumnRefSet(projectOperator.getColumnRefMap().keySet());
```

Comment:
How about trying to find a const col form the original projection firstly, if not, then trying to find the intersection?